cockatrice/src/dlg_connect.cpp

```diff
@@ -156,26 +165,104 @@ void DlgConnect::actSaveConfig()
                                               passwordEdit->text(), savePasswordCheckBox->isChecked());
     }
 
-    rebuildComboBoxList();
+    preRebuildComboBoxList();
 }
 
-void DlgConnect::rebuildComboBoxList()
+void DlgConnect::downloadPublicServers()
 {
-    previousHosts->clear();
+    QUrl url(QString(PUBLIC_SERVERS_URL));
+    auto *nam = new QNetworkAccessManager(this);
+    QNetworkReply *reply = nam->get(QNetworkRequest(url));
+    connect(reply, SIGNAL(finished()), this, SLOT(actFinishParsingDownloadedData()));
+}
+
+void DlgConnect::actFinishParsingDownloadedData()
+{
+    auto *reply = dynamic_cast<QNetworkReply *>(sender());
+    QNetworkReply::NetworkError errorCode = reply->error();
+
+    if (errorCode == QNetworkReply::NoError) {
+        UserConnection_Information uci;
+        savedHostList = uci.getServerInfo();
+
+        QStringList serversOnWiki;
+
+        // This will get all the data from the GitHub page and just give us the table data
+        QString stringData =
+            QString(reply->readAll()).remove(QRegExp(R"([\n\t\r])")).split("<tbody>").at(1).split("</tbody>").at(0);
+
+        // [Server1 Details], [Server2 Details], ...
+        QStringList serverList = stringData.split(QRegExp("<tr[^>]*>"));
+        serverList.removeAt(0);
+
+        for (const auto &server : serverList) {
+            QStringList serverValues = server.split(QRegExp("<td[^>]*>"));
+            QString serverName, serverAddress, serverPort;
+
+            for (int fieldCheck = 1; fieldCheck < serverValues.size(); fieldCheck++) {
+                // The fields correspond with the headers on the server list (index starting at 1)
+                switch (fieldCheck) {
+                    case 1: // Server Name
+                        serverName =
+                            QString(serverValues.at(fieldCheck)).remove(QRegExp("<a[^>]*>")).split("</a>").at(0);
+                        break;
+
+                    case 5: // Server Address
+                        serverAddress =
+                            QString(serverValues.at(fieldCheck)).remove("<strong>").split("</strong>").at(0);
+                        break;
+
+                    case 6: // Server Port
+                        serverPort = QString(serverValues.at(fieldCheck)).remove("<strong>").split("</strong>").at(0);
+                        break;
+
+                    default:
+                        break;
+                }
+            }
```

Comment:
I looping really necessary here? All it does is increment an index variable, which is used in a switch - wouldn't directly indexing the `serverValues` list be nicer?

```
short nameIndex = 1;
short addressIndex = 5;
short portIndex = 6;
serverName = QString(serverValues.at(nameIndex)).remove(QRegExp("<a[^>]*>")).split("</a>").at(0);
serverAddress = QString(serverValues.at(addressIndex)).remove("<strong>").split("</strong>").at(0);
serverPort = QString(serverValues.at(portIndex)).remove("<strong>").split("</strong>").at(0);
...
```

Reply:
Honestly, I wrote this on a whim and once it worked I put it up for review. Looking back, the loop is kind of pointless and I'm sure there's a better way

Reply:
By the way I don't really like that we get this list from a displayed source; if someone would change the formatting of the text (e.g. make it italic) the code would also have to be changed. It would be much better imo if we either had a dedicated file for the server list or used an API as @ancestral proposed.

Reply:
I'd be fine with having a dedicated file; where would we store it?